src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1125,22 +1141,19 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
             {
                 const auto* refInfo = consume<PoolIndex<FieldRefInfo>>(current).resolve(classFile);
 
-                llvm::Value* value = operandStack.back();
-                operandStack.pop_back();
-                llvm::Value* objectRef = operandStack.back();
-                operandStack.pop_back();
-
                 llvm::StringRef className = refInfo->classIndex.resolve(classFile)->nameIndex.resolve(classFile)->text;
                 llvm::StringRef fieldName =
                     refInfo->nameAndTypeIndex.resolve(classFile)->nameIndex.resolve(classFile)->text;
                 llvm::StringRef fieldType =
                     refInfo->nameAndTypeIndex.resolve(classFile)->descriptorIndex.resolve(classFile)->text;
+                llvm::Type* llvmFieldType = descriptorToType(parseFieldType(fieldType), builder.getContext());
+                llvm::Value* value = operandStack.pop_back(!llvmFieldType->isIntegerTy() || llvmFieldType->getIntegerBitWidth() >= 32 ? llvmFieldType : builder.getInt32Ty());
```

Comment:
Could you extract the `!llvmFieldType->isIntegerTy() || llvmFieldType->getIntegerBitWidth() >= 32 ? llvmFieldType : builder.getInt32Ty()` into a small function and use that here and below?